client/components/Run/index.jsx

```diff
@@ -1,34 +1,58 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import { Loader } from 'semantic-ui-react';
 import Scenario from '@components/Scenario';
+import { setRun } from '@client/actions';
 
 class Run extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            runId: undefined,
             scenarioId:
                 this.props.scenarioId || this.props.match.params.scenarioId,
             responses: new Map()
         };
 
+        this.onChange = this.onChange.bind(this);
         this.onResponseChange = this.onResponseChange.bind(this);
         this.onSubmit = this.onSubmit.bind(this);
+
+        this.fetchRun();
     }
 
-    async componentDidMount() {
-        const runData = await (await fetch(
-            `/api/runs/create/scenario/${this.state.scenarioId}`,
+    async fetchRun() {
+        const { run, status } = await (await fetch(
+            `/api/runs/new-or-existing/scenario/${this.state.scenarioId}`,
```

Comment:
I had considered using the word "upsert", but in a different way, however I decided against it because this isn't an "upsert", which is: "insert a new record, or update an existing". There is no "update an existing" here, it's just "return the existing record, or insert a new one".

Reply:
Ah okay, we can keep it then.

Reply:
Unless you like `/fetch/` better?